src/components/Dashboard/CountryList.js

```diff
@@ -7,20 +7,38 @@ import Country from './Country';
 import styles from './country.module.css';
 
 const CountryList = () => {
-  const countryList = useSelector((state) => state.pollutionSlice.countryList);
+  const state = useSelector((state) => state.pollutionSlice.countryList);
 
   const dispatch = useDispatch();
 
+  const [search, setSearch] = useState('');
+
+  const handleChange = (e) => {
+    setSearch(e.target.value);
+  };
+
   useEffect(() => {
-    if (!countryList) {
+    if (!state) {
       dispatch(fetchApi());
     }
-  }, [dispatch, countryList]);
+  }, [dispatch, state]);
+
+  const countryList = state.filter(
+    (item) => item.name.toLowerCase().includes(search.toLowerCase()),
+  );
```

Comment:
- [ ] Dear @Wahaj-Ali, You did well adding the required changes, But while doing so I notice you have introduced a bug into the project and I could not proceed to assert if the changes are implemented and your project is working fine. I have checked your project both on the `live` and `local` machines, the `dev` branch works fine but does not have the changes, and the `test` seems to be falling, I have attached the video and hope it will be helpful. 

> I also notice the exception is raised from this line. Please address this and make sure the required changes are in place.
 
### Test Brach 
https://user-images.githubusercontent.com/69077061/217799648-e99f4d11-deec-47af-b533-17d1d31fb5d0.mp4

### Dev branch 

https://user-images.githubusercontent.com/69077061/217799916-6ed325ec-295a-4657-b748-e9f60872f530.mp4

Reply:
@Tadesse-Alemayehu thank you for your suggestions. I want to mention that this PR is from the 'tests' branch to the 'dev' branch, so the dev branch does not have any latest changes. This is why the dev branch is working fine but does not contain the latest changes.

Reply:
Thanks for reaching out @Wahaj-Ali, the reason I refer to the Dev branch is that I notice the change the previous reviewer reviewed were already on the dev branch. As a pull request submitted for review, every change should be on the new branch including the initial functionalities.